sycl/test/regression/check_config_processing.cpp

```diff
@@ -0,0 +1,158 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+//
+// RUN: env TEST=SPACE_AT_FIRST_POSITION %t.out > %t1.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t1.conf %t.out
+//
+// RUN: env TEST=SPACE_AT_LAST_POSITION %t.out > %t2.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t2.conf %t.out
+//
+// RUN: env TEST=SPACE_BEFORE_ASSIGNMENT %t.out > %t3.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t3.conf %t.out
+//
+// RUN: env TEST=SPACE_AFTER_ASSIGNMENT %t.out > %t4.conf
+// RUN: env TEST=EXCESS_SPACE env SYCL_CONFIG_FILE_NAME=%t4.conf %t.out
+//
+// RUN: env TEST=VARIABLE_NAME_BIGGER_THAN_MAX_CONFIG_NAME %t.out > %t5.conf
+// RUN: env TEST=INCORRECT_VARIABLE_NAME env SYCL_CONFIG_FILE_NAME=%t5.conf %t.out
+//
+// RUN: env TEST=VARIABLE_WITHOUT_NAME %t.out > %t6.conf
+// RUN: env TEST=INCORRECT_VARIABLE_NAME env SYCL_CONFIG_FILE_NAME=%t6.conf %t.out
+//
+// RUN: env TEST=VARIABLE_VALUE_BIGGER_THAN_MAX_CONFIG_VALUE %t.out > %t7.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t7.conf %t.out
+//
+// RUN: env TEST=VARIABLE_WITHOUT_VALUE %t.out > %t8.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t8.conf %t.out
+//
+// RUN: env TEST=COMPLEX_CONFIG %t.out > %t9.conf
+// RUN: env TEST=INCORRECT_VARIABLE_VALUE env SYCL_CONFIG_FILE_NAME=%t9.conf %t.out
+//
+// RUN: env TEST=SET_CORRECT_ENVIRONMENT %t.out > %t10.conf
+// RUN: env TEST=CORRECT_CONFIG_FILE env SYCL_CONFIG_FILE_NAME=%t10.conf %t.out
+// RUN: ls | grep dot
+// RUN: rm *.dot
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <regex>
+
+int main() {
+  std::string testEnvVarValue;
+  if (getenv("TEST")) {
```

Comment:
Does the test verifies getenv() function or test itself? What is the reason to throw exception?

Reply:
In this line we check that environment TEST is exist. If we try to use getenv("TEST") without env TEST=..., then we will receive error in line with comparison (for example, testEnvVarValue == "SPACE_AT_FIRST_POSITION") because in testEnvVarValue will be a null pointer, which can not be compared with the string. (We throw an exception because we expect to run tests with the environment)

Reply:
it looks like a test which checks itself. For me this check is redundant.

Reply:
I can remove "if (getenv("TEST"))", but that might confuse users, which will try to run this tests without required environment.

Reply:
ok